Widgets/AssetView.h

```diff
@@ -1,23 +1,41 @@
 #ifndef ASSETVIEW_H
 #define ASSETVIEW_H
 
+#include "AssetScrollAreaBackground.h"
+
 #include <QPainter>
 #include <QWidget>
 
+enum GridType {
+  Standard,   // Simple x by x Grid
+  Complex,    // Grid with offset and spacing between squares
+  Isometric,  // TODO: Implement this
+};
+
+struct GridDimensions {
+  GridType type = GridType::Standard;
+  bool show = true;
+  int horSpacing = 16;
+  int vertSpacing = 16;
+  int horOff = 0;
+  int vertOff = 0;
+  int cellWidth = 0;
+  int cellHeight = 0;

```

Comment:
Another idea is to define this as a unitCell rectangle, capturing horSpacing, vertSpacing, cellWidth, and cellHeight. The grid is then simply a tiled matrix of unit cells.

Reply:
I don't feel like breaking that function atm.